From 6b3ef6516cb33cb45dc08eb2d0646ba62d6fa870 Mon Sep 17 00:00:00 2001
From: WofWca <wofwca@protonmail.com>
Date: Sun, 9 Aug 2026 22:00:08 +0400
Subject: [PATCH] refactor: add _description param to *G_CVAR macros

Unused as of writing, but in preparation for the potential future.
---
 code/cgame/cg_cvar.h | 212 +++++++++++++++++++++----------------------
 code/game/g_cvar.h   | 126 ++++++++++++-------------
 2 files changed, 169 insertions(+), 169 deletions(-)

diff --git a/code/cgame/cg_cvar.h b/code/cgame/cg_cvar.h
index 67a4e3e39..f2af7847a 100644
--- a/code/cgame/cg_cvar.h
+++ b/code/cgame/cg_cvar.h
@@ -23,134 +23,134 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
 #ifdef CG_CVAR_TABLE
-#define CG_CVAR( var, name, defaultVal, flags ) \
+#define CG_CVAR( var, name, defaultVal, flags, _description ) \
 	{ &var, name, defaultVal, flags },
 #endif
 
 #ifdef CG_CVAR_DECLARATION
-#define CG_CVAR( var, name, defaultVal, flags ) \
+#define CG_CVAR( var, name, defaultVal, flags, _description ) \
 	vmCvar_t	var;
 #endif
 
 #ifdef CG_CVAR_EXTERN
-#define CG_CVAR( var, name, defaultVal, flags ) \
+#define CG_CVAR( var, name, defaultVal, flags, _description ) \
 	extern	vmCvar_t	var;
 #endif
 
-CG_CVAR( cg_ignore, "cg_ignore", "0", 0 )	// used for debugging
-CG_CVAR( cg_autoswitch, "cg_autoswitch", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawGun, "cg_drawGun", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_zoomFov, "cg_zoomfov", "22.5", CVAR_ARCHIVE )
-CG_CVAR( cg_fov, "cg_fov", "90", CVAR_ARCHIVE )
-CG_CVAR( cg_viewsize, "cg_viewsize", "100", CVAR_ARCHIVE )
-CG_CVAR( cg_shadows, "cg_shadows", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_gibs, "cg_gibs", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_draw2D, "cg_draw2D", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawStatus, "cg_drawStatus", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawTimer, "cg_drawTimer", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_drawFPS, "cg_drawFPS", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_drawSnapshot, "cg_drawSnapshot", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_draw3dIcons, "cg_draw3dIcons", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawIcons, "cg_drawIcons", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawAmmoWarning, "cg_drawAmmoWarning", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawAttacker, "cg_drawAttacker", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawCrosshair, "cg_drawCrosshair", "4", CVAR_ARCHIVE )
-CG_CVAR( cg_drawCrosshairNames, "cg_drawCrosshairNames", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_drawRewards, "cg_drawRewards", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_crosshairSize, "cg_crosshairSize", "24", CVAR_ARCHIVE )
-CG_CVAR( cg_crosshairHealth, "cg_crosshairHealth", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_crosshairX, "cg_crosshairX", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_crosshairY, "cg_crosshairY", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_brassTime, "cg_brassTime", "2500", CVAR_ARCHIVE )
-CG_CVAR( cg_simpleItems, "cg_simpleItems", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_addMarks, "cg_marks", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_lagometer, "cg_lagometer", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_railTrailTime, "cg_railTrailTime", "400", CVAR_ARCHIVE )
-// CG_CVAR( cg_gun_frame, "cg_gun_frame", "", CVAR_CHEAT )
-CG_CVAR( cg_gun_x, "cg_gunX", "0", CVAR_CHEAT )
-CG_CVAR( cg_gun_y, "cg_gunY", "0", CVAR_CHEAT )
-CG_CVAR( cg_gun_z, "cg_gunZ", "0", CVAR_CHEAT )
-CG_CVAR( cg_centertime, "cg_centertime", "3", CVAR_CHEAT )
-CG_CVAR( cg_runpitch, "cg_runpitch", "0.002", CVAR_ARCHIVE )
-CG_CVAR( cg_runroll, "cg_runroll", "0.005", CVAR_ARCHIVE )
-CG_CVAR( cg_bobup, "cg_bobup", "0.005", CVAR_CHEAT )
-CG_CVAR( cg_bobpitch, "cg_bobpitch", "0.002", CVAR_ARCHIVE )
-CG_CVAR( cg_bobroll, "cg_bobroll", "0.002", CVAR_ARCHIVE )
-CG_CVAR( cg_swingSpeed, "cg_swingSpeed", "0.3", CVAR_CHEAT )
-CG_CVAR( cg_animSpeed, "cg_animspeed", "1", CVAR_CHEAT )
-CG_CVAR( cg_debugAnim, "cg_debuganim", "0", CVAR_CHEAT )
-CG_CVAR( cg_debugPosition, "cg_debugposition", "0", CVAR_CHEAT )
-CG_CVAR( cg_debugEvents, "cg_debugevents", "0", CVAR_CHEAT )
-CG_CVAR( cg_errorDecay, "cg_errordecay", "100", 0 )
-CG_CVAR( cg_nopredict, "cg_nopredict", "0", 0 )
-CG_CVAR( cg_noPlayerAnims, "cg_noplayeranims", "0", CVAR_CHEAT )
-CG_CVAR( cg_showmiss, "cg_showmiss", "0", 0 )
-CG_CVAR( cg_footsteps, "cg_footsteps", "1", CVAR_CHEAT )
-CG_CVAR( cg_tracerChance, "cg_tracerchance", "0.4", CVAR_CHEAT )
-CG_CVAR( cg_tracerWidth, "cg_tracerwidth", "1", CVAR_CHEAT )
-CG_CVAR( cg_tracerLength, "cg_tracerlength", "100", CVAR_CHEAT )
-CG_CVAR( cg_thirdPersonRange, "cg_thirdPersonRange", "40", CVAR_CHEAT )
-CG_CVAR( cg_thirdPersonAngle, "cg_thirdPersonAngle", "0", CVAR_CHEAT )
-CG_CVAR( cg_thirdPerson, "cg_thirdPerson", "0", 0 )
-CG_CVAR( cg_teamChatTime, "cg_teamChatTime", "3000", CVAR_ARCHIVE )
-CG_CVAR( cg_teamChatHeight, "cg_teamChatHeight", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_forceModel, "cg_forceModel", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_predictItems, "cg_predictItems", "1", CVAR_ARCHIVE )
+CG_CVAR( cg_ignore, "cg_ignore", "0", 0, NULL )	// used for debugging
+CG_CVAR( cg_autoswitch, "cg_autoswitch", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawGun, "cg_drawGun", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_zoomFov, "cg_zoomfov", "22.5", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_fov, "cg_fov", "90", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_viewsize, "cg_viewsize", "100", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_shadows, "cg_shadows", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_gibs, "cg_gibs", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_draw2D, "cg_draw2D", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawStatus, "cg_drawStatus", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawTimer, "cg_drawTimer", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawFPS, "cg_drawFPS", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawSnapshot, "cg_drawSnapshot", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_draw3dIcons, "cg_draw3dIcons", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawIcons, "cg_drawIcons", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawAmmoWarning, "cg_drawAmmoWarning", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawAttacker, "cg_drawAttacker", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawCrosshair, "cg_drawCrosshair", "4", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawCrosshairNames, "cg_drawCrosshairNames", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_drawRewards, "cg_drawRewards", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_crosshairSize, "cg_crosshairSize", "24", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_crosshairHealth, "cg_crosshairHealth", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_crosshairX, "cg_crosshairX", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_crosshairY, "cg_crosshairY", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_brassTime, "cg_brassTime", "2500", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_simpleItems, "cg_simpleItems", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_addMarks, "cg_marks", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_lagometer, "cg_lagometer", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_railTrailTime, "cg_railTrailTime", "400", CVAR_ARCHIVE, NULL )
+// CG_CVAR( cg_gun_frame, "cg_gun_frame", "", CVAR_CHEAT, NULL )
+CG_CVAR( cg_gun_x, "cg_gunX", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_gun_y, "cg_gunY", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_gun_z, "cg_gunZ", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_centertime, "cg_centertime", "3", CVAR_CHEAT, NULL )
+CG_CVAR( cg_runpitch, "cg_runpitch", "0.002", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_runroll, "cg_runroll", "0.005", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_bobup, "cg_bobup", "0.005", CVAR_CHEAT, NULL )
+CG_CVAR( cg_bobpitch, "cg_bobpitch", "0.002", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_bobroll, "cg_bobroll", "0.002", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_swingSpeed, "cg_swingSpeed", "0.3", CVAR_CHEAT, NULL )
+CG_CVAR( cg_animSpeed, "cg_animspeed", "1", CVAR_CHEAT, NULL )
+CG_CVAR( cg_debugAnim, "cg_debuganim", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_debugPosition, "cg_debugposition", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_debugEvents, "cg_debugevents", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_errorDecay, "cg_errordecay", "100", 0, NULL )
+CG_CVAR( cg_nopredict, "cg_nopredict", "0", 0, NULL )
+CG_CVAR( cg_noPlayerAnims, "cg_noplayeranims", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_showmiss, "cg_showmiss", "0", 0, NULL )
+CG_CVAR( cg_footsteps, "cg_footsteps", "1", CVAR_CHEAT, NULL )
+CG_CVAR( cg_tracerChance, "cg_tracerchance", "0.4", CVAR_CHEAT, NULL )
+CG_CVAR( cg_tracerWidth, "cg_tracerwidth", "1", CVAR_CHEAT, NULL )
+CG_CVAR( cg_tracerLength, "cg_tracerlength", "100", CVAR_CHEAT, NULL )
+CG_CVAR( cg_thirdPersonRange, "cg_thirdPersonRange", "40", CVAR_CHEAT, NULL )
+CG_CVAR( cg_thirdPersonAngle, "cg_thirdPersonAngle", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_thirdPerson, "cg_thirdPerson", "0", 0, NULL )
+CG_CVAR( cg_teamChatTime, "cg_teamChatTime", "3000", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_teamChatHeight, "cg_teamChatHeight", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_forceModel, "cg_forceModel", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_predictItems, "cg_predictItems", "1", CVAR_ARCHIVE, NULL )
 #ifdef MISSIONPACK
-CG_CVAR( cg_deferPlayers, "cg_deferPlayers", "0", CVAR_ARCHIVE )
+CG_CVAR( cg_deferPlayers, "cg_deferPlayers", "0", CVAR_ARCHIVE, NULL )
 #else
-CG_CVAR( cg_deferPlayers, "cg_deferPlayers", "1", CVAR_ARCHIVE )
+CG_CVAR( cg_deferPlayers, "cg_deferPlayers", "1", CVAR_ARCHIVE, NULL )
 #endif
-CG_CVAR( cg_drawTeamOverlay, "cg_drawTeamOverlay", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_teamOverlayUserinfo, "teamoverlay", "0", CVAR_ROM | CVAR_USERINFO )
-CG_CVAR( cg_stats, "cg_stats", "0", 0 )
-CG_CVAR( cg_drawFriend, "cg_drawFriend", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_teamChatsOnly, "cg_teamChatsOnly", "0", CVAR_ARCHIVE )
+CG_CVAR( cg_drawTeamOverlay, "cg_drawTeamOverlay", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_teamOverlayUserinfo, "teamoverlay", "0", CVAR_ROM | CVAR_USERINFO, NULL )
+CG_CVAR( cg_stats, "cg_stats", "0", 0, NULL )
+CG_CVAR( cg_drawFriend, "cg_drawFriend", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_teamChatsOnly, "cg_teamChatsOnly", "0", CVAR_ARCHIVE, NULL )
 #ifdef MISSIONPACK
-CG_CVAR( cg_noVoiceChats, "cg_noVoiceChats", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_noVoiceText, "cg_noVoiceText", "0", CVAR_ARCHIVE )
+CG_CVAR( cg_noVoiceChats, "cg_noVoiceChats", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_noVoiceText, "cg_noVoiceText", "0", CVAR_ARCHIVE, NULL )
 #endif
 // the following variables are created in other parts of the system,
 // but we also reference them here
-CG_CVAR( cg_buildScript, "com_buildScript", "0", 0 )	// force loading of all possible data amd error on failures
-CG_CVAR( cg_paused, "cl_paused", "0", CVAR_ROM )
-CG_CVAR( cg_blood, "com_blood", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_synchronousClients, "g_synchronousClients", "0", CVAR_SYSTEMINFO )
+CG_CVAR( cg_buildScript, "com_buildScript", "0", 0, NULL )	// force loading of all possible data amd error on failures
+CG_CVAR( cg_paused, "cl_paused", "0", CVAR_ROM, NULL )
+CG_CVAR( cg_blood, "com_blood", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_synchronousClients, "g_synchronousClients", "0", CVAR_SYSTEMINFO, NULL )
 #ifdef MISSIONPACK
-CG_CVAR( cg_redTeamName, "g_redteam", DEFAULT_REDTEAM_NAME, CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO )
-CG_CVAR( cg_blueTeamName, "g_blueteam", DEFAULT_BLUETEAM_NAME, CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO )
-CG_CVAR( cg_currentSelectedPlayer, "cg_currentSelectedPlayer", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_currentSelectedPlayerName, "cg_currentSelectedPlayerName", "", CVAR_ARCHIVE )
-CG_CVAR( cg_singlePlayer, "ui_singlePlayerActive", "0", CVAR_USERINFO )
-CG_CVAR( cg_enableDust, "g_enableDust", "0", CVAR_SERVERINFO )
-CG_CVAR( cg_enableBreath, "g_enableBreath", "0", CVAR_SERVERINFO )
-CG_CVAR( cg_singlePlayerActive, "ui_singlePlayerActive", "0", CVAR_USERINFO )
-CG_CVAR( cg_recordSPDemo, "ui_recordSPDemo", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_recordSPDemoName, "ui_recordSPDemoName", "", CVAR_ARCHIVE )
-CG_CVAR( cg_obeliskRespawnDelay, "g_obeliskRespawnDelay", "10", CVAR_SERVERINFO )
-CG_CVAR( cg_hudFiles, "cg_hudFiles", "ui/hud.txt", CVAR_ARCHIVE )
+CG_CVAR( cg_redTeamName, "g_redteam", DEFAULT_REDTEAM_NAME, CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO, NULL )
+CG_CVAR( cg_blueTeamName, "g_blueteam", DEFAULT_BLUETEAM_NAME, CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO, NULL )
+CG_CVAR( cg_currentSelectedPlayer, "cg_currentSelectedPlayer", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_currentSelectedPlayerName, "cg_currentSelectedPlayerName", "", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_singlePlayer, "ui_singlePlayerActive", "0", CVAR_USERINFO, NULL )
+CG_CVAR( cg_enableDust, "g_enableDust", "0", CVAR_SERVERINFO, NULL )
+CG_CVAR( cg_enableBreath, "g_enableBreath", "0", CVAR_SERVERINFO, NULL )
+CG_CVAR( cg_singlePlayerActive, "ui_singlePlayerActive", "0", CVAR_USERINFO, NULL )
+CG_CVAR( cg_recordSPDemo, "ui_recordSPDemo", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_recordSPDemoName, "ui_recordSPDemoName", "", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_obeliskRespawnDelay, "g_obeliskRespawnDelay", "10", CVAR_SERVERINFO, NULL )
+CG_CVAR( cg_hudFiles, "cg_hudFiles", "ui/hud.txt", CVAR_ARCHIVE, NULL )
 #endif
-CG_CVAR( cg_cameraOrbit, "cg_cameraOrbit", "0", CVAR_CHEAT )
-CG_CVAR( cg_cameraOrbitDelay, "cg_cameraOrbitDelay", "50", CVAR_ARCHIVE )
-CG_CVAR( cg_timescaleFadeEnd, "cg_timescaleFadeEnd", "1", 0 )
-CG_CVAR( cg_timescaleFadeSpeed, "cg_timescaleFadeSpeed", "0", 0 )
-CG_CVAR( cg_timescale, "timescale", "1", 0 )
-CG_CVAR( cg_scorePlum, "cg_scorePlums", "1", CVAR_USERINFO | CVAR_ARCHIVE )
-CG_CVAR( cg_smoothClients, "cg_smoothClients", "0", CVAR_USERINFO | CVAR_ARCHIVE )
-CG_CVAR( cg_cameraMode, "com_cameraMode", "0", CVAR_CHEAT )
+CG_CVAR( cg_cameraOrbit, "cg_cameraOrbit", "0", CVAR_CHEAT, NULL )
+CG_CVAR( cg_cameraOrbitDelay, "cg_cameraOrbitDelay", "50", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_timescaleFadeEnd, "cg_timescaleFadeEnd", "1", 0, NULL )
+CG_CVAR( cg_timescaleFadeSpeed, "cg_timescaleFadeSpeed", "0", 0, NULL )
+CG_CVAR( cg_timescale, "timescale", "1", 0, NULL )
+CG_CVAR( cg_scorePlum, "cg_scorePlums", "1", CVAR_USERINFO | CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_smoothClients, "cg_smoothClients", "0", CVAR_USERINFO | CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_cameraMode, "com_cameraMode", "0", CVAR_CHEAT, NULL )
 
-CG_CVAR( pmove_fixed, "pmove_fixed", "0", CVAR_SYSTEMINFO )
-CG_CVAR( pmove_msec, "pmove_msec", "8", CVAR_SYSTEMINFO )
+CG_CVAR( pmove_fixed, "pmove_fixed", "0", CVAR_SYSTEMINFO, NULL )
+CG_CVAR( pmove_msec, "pmove_msec", "8", CVAR_SYSTEMINFO, NULL )
 #ifdef MISSIONPACK
-CG_CVAR( cg_smallFont, "ui_smallFont", "0.25", CVAR_ARCHIVE )
-CG_CVAR( cg_bigFont, "ui_bigFont", "0.4", CVAR_ARCHIVE )
-CG_CVAR( cg_noTaunt, "cg_noTaunt", "0", CVAR_ARCHIVE )
+CG_CVAR( cg_smallFont, "ui_smallFont", "0.25", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_bigFont, "ui_bigFont", "0.4", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_noTaunt, "cg_noTaunt", "0", CVAR_ARCHIVE, NULL )
 #endif
-CG_CVAR( cg_noProjectileTrail, "cg_noProjectileTrail", "0", CVAR_ARCHIVE )
-CG_CVAR( cg_oldRail, "cg_oldRail", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_oldRocket, "cg_oldRocket", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_oldPlasma, "cg_oldPlasma", "1", CVAR_ARCHIVE )
-CG_CVAR( cg_trueLightning, "cg_trueLightning", "0.0", CVAR_ARCHIVE )
-// CG_CVAR( cg_pmove_fixed, "cg_pmove_fixed", "0", CVAR_USERINFO | CVAR_ARCHIVE )
+CG_CVAR( cg_noProjectileTrail, "cg_noProjectileTrail", "0", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_oldRail, "cg_oldRail", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_oldRocket, "cg_oldRocket", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_oldPlasma, "cg_oldPlasma", "1", CVAR_ARCHIVE, NULL )
+CG_CVAR( cg_trueLightning, "cg_trueLightning", "0.0", CVAR_ARCHIVE, NULL )
+// CG_CVAR( cg_pmove_fixed, "cg_pmove_fixed", "0", CVAR_USERINFO | CVAR_ARCHIVE, NULL )
 
 #undef CG_CVAR
diff --git a/code/game/g_cvar.h b/code/game/g_cvar.h
index ba727f9f8..a2e642782 100644
--- a/code/game/g_cvar.h
+++ b/code/game/g_cvar.h
@@ -23,23 +23,23 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
 #ifdef G_CVAR_TABLE
-#define G_CVAR( var, name, defaultVal, flags, trackChange ) \
+#define G_CVAR( var, name, defaultVal, flags, trackChange, _description ) \
 	{ &var, name, defaultVal, flags, 0, trackChange, qfalse },
-#define G_CVAR_EXT( var, name, defaultVal, flags, modificationCount, trackChange, teamShader ) \
+#define G_CVAR_EXT( var, name, defaultVal, flags, modificationCount, trackChange, teamShader, _description ) \
 	{ &var, name, defaultVal, flags, modificationCount, trackChange, teamShader },
 #endif
 
 #ifdef G_CVAR_DECLARATION
-#define G_CVAR( var, name, defaultVal, flags, trackChange ) \
+#define G_CVAR( var, name, defaultVal, flags, trackChange, _description ) \
 	vmCvar_t	var;
-#define G_CVAR_EXT( var, name, defaultVal, flags, modificationCount, trackChange, teamShader ) \
+#define G_CVAR_EXT( var, name, defaultVal, flags, modificationCount, trackChange, teamShader, _description ) \
 	vmCvar_t	var;
 #endif
 
 #ifdef G_CVAR_EXTERN
-#define G_CVAR( var, name, defaultVal, flags, trackChange ) \
+#define G_CVAR( var, name, defaultVal, flags, trackChange, _description ) \
 	extern	vmCvar_t	var;
-#define G_CVAR_EXT( var, name, defaultVal, flags, modificationCount, trackChange, teamShader ) \
+#define G_CVAR_EXT( var, name, defaultVal, flags, modificationCount, trackChange, teamShader, _description ) \
 	extern	vmCvar_t	var;
 #endif
 
@@ -50,86 +50,86 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #define NO_TRACK qfalse
 
 // don't override the cheat state set by the system
-G_CVAR( g_cheats, "sv_cheats", "", NO_FLAGS, NO_TRACK )
+G_CVAR( g_cheats, "sv_cheats", "", NO_FLAGS, NO_TRACK, NULL )
 
-G_CVAR( g_restarted, "g_restarted", "0", CVAR_ROM, NO_TRACK )
+G_CVAR( g_restarted, "g_restarted", "0", CVAR_ROM, NO_TRACK, NULL )
 
 // latched vars
-G_CVAR( g_gametype, "g_gametype", "0", CVAR_SERVERINFO | CVAR_USERINFO | CVAR_LATCH, NO_TRACK )
+G_CVAR( g_gametype, "g_gametype", "0", CVAR_SERVERINFO | CVAR_USERINFO | CVAR_LATCH, NO_TRACK, NULL )
 
 // allow this many total, including spectators
-G_CVAR( g_maxclients, "sv_maxclients", "8", CVAR_SERVERINFO | CVAR_LATCH | CVAR_ARCHIVE, NO_TRACK )
+G_CVAR( g_maxclients, "sv_maxclients", "8", CVAR_SERVERINFO | CVAR_LATCH | CVAR_ARCHIVE, NO_TRACK, NULL )
 // allow this many active
-G_CVAR( g_maxGameClients, "g_maxGameClients", "0", CVAR_SERVERINFO | CVAR_LATCH | CVAR_ARCHIVE, NO_TRACK )
+G_CVAR( g_maxGameClients, "g_maxGameClients", "0", CVAR_SERVERINFO | CVAR_LATCH | CVAR_ARCHIVE, NO_TRACK, NULL )
 
 // change anytime vars
-G_CVAR( g_dmflags, "dmflags", "0", CVAR_SERVERINFO | CVAR_ARCHIVE, TRACK )
-G_CVAR( g_fraglimit, "fraglimit", "20", CVAR_SERVERINFO | CVAR_ARCHIVE | CVAR_NORESTART, TRACK )
-G_CVAR( g_timelimit, "timelimit", "0", CVAR_SERVERINFO | CVAR_ARCHIVE | CVAR_NORESTART, TRACK )
-G_CVAR( g_capturelimit, "capturelimit", "8", CVAR_SERVERINFO | CVAR_ARCHIVE | CVAR_NORESTART, TRACK )
+G_CVAR( g_dmflags, "dmflags", "0", CVAR_SERVERINFO | CVAR_ARCHIVE, TRACK, NULL )
+G_CVAR( g_fraglimit, "fraglimit", "20", CVAR_SERVERINFO | CVAR_ARCHIVE | CVAR_NORESTART, TRACK, NULL )
+G_CVAR( g_timelimit, "timelimit", "0", CVAR_SERVERINFO | CVAR_ARCHIVE | CVAR_NORESTART, TRACK, NULL )
+G_CVAR( g_capturelimit, "capturelimit", "8", CVAR_SERVERINFO | CVAR_ARCHIVE | CVAR_NORESTART, TRACK, NULL )
 
-G_CVAR( g_synchronousClients, "g_synchronousClients", "0", CVAR_SYSTEMINFO, NO_TRACK )
+G_CVAR( g_synchronousClients, "g_synchronousClients", "0", CVAR_SYSTEMINFO, NO_TRACK, NULL )
 
-G_CVAR( g_friendlyFire, "g_friendlyFire", "0", CVAR_ARCHIVE, TRACK )
+G_CVAR( g_friendlyFire, "g_friendlyFire", "0", CVAR_ARCHIVE, TRACK, NULL )
 
-G_CVAR( g_teamAutoJoin, "g_teamAutoJoin", "0", CVAR_ARCHIVE, NO_TRACK )
-G_CVAR( g_teamForceBalance, "g_teamForceBalance", "0", CVAR_ARCHIVE, NO_TRACK )
+G_CVAR( g_teamAutoJoin, "g_teamAutoJoin", "0", CVAR_ARCHIVE, NO_TRACK, NULL )
+G_CVAR( g_teamForceBalance, "g_teamForceBalance", "0", CVAR_ARCHIVE, NO_TRACK, NULL )
 
-G_CVAR( g_warmup, "g_warmup", "20", CVAR_ARCHIVE, TRACK )
-G_CVAR( g_doWarmup, "g_doWarmup", "0", CVAR_ARCHIVE, TRACK )
-G_CVAR( g_logfile, "g_log", "games.log", CVAR_ARCHIVE, NO_TRACK )
-G_CVAR( g_logfileSync, "g_logsync", "0", CVAR_ARCHIVE, NO_TRACK )
+G_CVAR( g_warmup, "g_warmup", "20", CVAR_ARCHIVE, TRACK, NULL )
+G_CVAR( g_doWarmup, "g_doWarmup", "0", CVAR_ARCHIVE, TRACK, NULL )
+G_CVAR( g_logfile, "g_log", "games.log", CVAR_ARCHIVE, NO_TRACK, NULL )
+G_CVAR( g_logfileSync, "g_logsync", "0", CVAR_ARCHIVE, NO_TRACK, NULL )
 
-G_CVAR( g_password, "g_password", "", CVAR_USERINFO, NO_TRACK )
+G_CVAR( g_password, "g_password", "", CVAR_USERINFO, NO_TRACK, NULL )
 
-G_CVAR( g_banIPs, "g_banIPs", "", CVAR_ARCHIVE, NO_TRACK )
-G_CVAR( g_filterBan, "g_filterBan", "1", CVAR_ARCHIVE, NO_TRACK )
+G_CVAR( g_banIPs, "g_banIPs", "", CVAR_ARCHIVE, NO_TRACK, NULL )
+G_CVAR( g_filterBan, "g_filterBan", "1", CVAR_ARCHIVE, NO_TRACK, NULL )
 
-G_CVAR( g_needpass, "g_needpass", "0", CVAR_SERVERINFO | CVAR_ROM, NO_TRACK )
+G_CVAR( g_needpass, "g_needpass", "0", CVAR_SERVERINFO | CVAR_ROM, NO_TRACK, NULL )
 
-G_CVAR( g_dedicated, "dedicated", "0", NO_FLAGS, NO_TRACK )
+G_CVAR( g_dedicated, "dedicated", "0", NO_FLAGS, NO_TRACK, NULL )
 
-G_CVAR( g_speed, "g_speed", "320", NO_FLAGS, TRACK )
-G_CVAR( g_gravity, "g_gravity", "800", NO_FLAGS, TRACK )
-G_CVAR( g_knockback, "g_knockback", "1000", NO_FLAGS, TRACK )
-G_CVAR( g_quadfactor, "g_quadfactor", "3", NO_FLAGS, TRACK )
-G_CVAR( g_weaponRespawn, "g_weaponrespawn", "5", NO_FLAGS, TRACK )
-G_CVAR( g_weaponTeamRespawn, "g_weaponTeamRespawn", "30", NO_FLAGS, TRACK )
-G_CVAR( g_forcerespawn, "g_forcerespawn", "20", NO_FLAGS, TRACK )
-G_CVAR( g_inactivity, "g_inactivity", "0", NO_FLAGS, TRACK )
-G_CVAR( g_debugMove, "g_debugMove", "0", NO_FLAGS, NO_TRACK )
-G_CVAR( g_debugDamage, "g_debugDamage", "0", NO_FLAGS, NO_TRACK )
-G_CVAR( g_debugAlloc, "g_debugAlloc", "0", NO_FLAGS, NO_TRACK )
-G_CVAR( g_motd, "g_motd", "", NO_FLAGS, NO_TRACK )
-G_CVAR( g_blood, "com_blood", "1", NO_FLAGS, NO_TRACK )
+G_CVAR( g_speed, "g_speed", "320", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_gravity, "g_gravity", "800", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_knockback, "g_knockback", "1000", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_quadfactor, "g_quadfactor", "3", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_weaponRespawn, "g_weaponrespawn", "5", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_weaponTeamRespawn, "g_weaponTeamRespawn", "30", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_forcerespawn, "g_forcerespawn", "20", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_inactivity, "g_inactivity", "0", NO_FLAGS, TRACK, NULL )
+G_CVAR( g_debugMove, "g_debugMove", "0", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_debugDamage, "g_debugDamage", "0", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_debugAlloc, "g_debugAlloc", "0", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_motd, "g_motd", "", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_blood, "com_blood", "1", NO_FLAGS, NO_TRACK, NULL )
 
-G_CVAR( g_podiumDist, "g_podiumDist", "80", NO_FLAGS, NO_TRACK )
-G_CVAR( g_podiumDrop, "g_podiumDrop", "70", NO_FLAGS, NO_TRACK )
+G_CVAR( g_podiumDist, "g_podiumDist", "80", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_podiumDrop, "g_podiumDrop", "70", NO_FLAGS, NO_TRACK, NULL )
 
-G_CVAR( g_allowVote, "g_allowVote", "1", CVAR_ARCHIVE, NO_TRACK )
-G_CVAR( g_listEntity, "g_listEntity", "0", NO_FLAGS, NO_TRACK )
+G_CVAR( g_allowVote, "g_allowVote", "1", CVAR_ARCHIVE, NO_TRACK, NULL )
+G_CVAR( g_listEntity, "g_listEntity", "0", NO_FLAGS, NO_TRACK, NULL )
 
 #ifdef MISSIONPACK
-G_CVAR( g_obeliskHealth, "g_obeliskHealth", "2500", NO_FLAGS, NO_TRACK )
-G_CVAR( g_obeliskRegenPeriod, "g_obeliskRegenPeriod", "1", NO_FLAGS, NO_TRACK )
-G_CVAR( g_obeliskRegenAmount, "g_obeliskRegenAmount", "15", NO_FLAGS, NO_TRACK )
-G_CVAR( g_obeliskRespawnDelay, "g_obeliskRespawnDelay", "10", CVAR_SERVERINFO, NO_TRACK )
-
-G_CVAR( g_cubeTimeout, "g_cubeTimeout", "30", NO_FLAGS, NO_TRACK )
-G_CVAR_EXT( g_redteam, "g_redteam", "Stroggs", CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO , 0, TRACK, qtrue )
-G_CVAR_EXT( g_blueteam, "g_blueteam", "Pagans", CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO , 0, TRACK, qtrue )
-G_CVAR( g_singlePlayer, "ui_singlePlayerActive", "", NO_FLAGS, NO_TRACK )
-
-G_CVAR( g_enableDust, "g_enableDust", "0", CVAR_SERVERINFO, TRACK )
-G_CVAR( g_enableBreath, "g_enableBreath", "0", CVAR_SERVERINFO, TRACK )
-G_CVAR( g_proxMineTimeout, "g_proxMineTimeout", "20000", NO_FLAGS, NO_TRACK )
+G_CVAR( g_obeliskHealth, "g_obeliskHealth", "2500", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_obeliskRegenPeriod, "g_obeliskRegenPeriod", "1", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_obeliskRegenAmount, "g_obeliskRegenAmount", "15", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_obeliskRespawnDelay, "g_obeliskRespawnDelay", "10", CVAR_SERVERINFO, NO_TRACK, NULL )
+
+G_CVAR( g_cubeTimeout, "g_cubeTimeout", "30", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR_EXT( g_redteam, "g_redteam", "Stroggs", CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO , 0, TRACK, qtrue, NULL )
+G_CVAR_EXT( g_blueteam, "g_blueteam", "Pagans", CVAR_ARCHIVE | CVAR_SERVERINFO | CVAR_USERINFO , 0, TRACK, qtrue, NULL )
+G_CVAR( g_singlePlayer, "ui_singlePlayerActive", "", NO_FLAGS, NO_TRACK, NULL )
+
+G_CVAR( g_enableDust, "g_enableDust", "0", CVAR_SERVERINFO, TRACK, NULL )
+G_CVAR( g_enableBreath, "g_enableBreath", "0", CVAR_SERVERINFO, TRACK, NULL )
+G_CVAR( g_proxMineTimeout, "g_proxMineTimeout", "20000", NO_FLAGS, NO_TRACK, NULL )
 #endif
-G_CVAR( g_smoothClients, "g_smoothClients", "1", NO_FLAGS, NO_TRACK )
-G_CVAR( pmove_fixed, "pmove_fixed", "0", CVAR_SYSTEMINFO, NO_TRACK )
-G_CVAR( pmove_msec, "pmove_msec", "8", CVAR_SYSTEMINFO, NO_TRACK )
+G_CVAR( g_smoothClients, "g_smoothClients", "1", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( pmove_fixed, "pmove_fixed", "0", CVAR_SYSTEMINFO, NO_TRACK, NULL )
+G_CVAR( pmove_msec, "pmove_msec", "8", CVAR_SYSTEMINFO, NO_TRACK, NULL )
 
-G_CVAR( g_rankings, "g_rankings", "0", NO_FLAGS, NO_TRACK )
-G_CVAR( g_localTeamPref, "g_localTeamPref", "", NO_FLAGS, NO_TRACK )
+G_CVAR( g_rankings, "g_rankings", "0", NO_FLAGS, NO_TRACK, NULL )
+G_CVAR( g_localTeamPref, "g_localTeamPref", "", NO_FLAGS, NO_TRACK, NULL )
 
 #undef G_CVAR
 #undef G_CVAR_EXT
-- 
2.55.0

